contentgrid-appserver-rest/src/main/java/com/contentgrid/appserver/rest/exception/InvalidRelationTargetException.java

```diff
@@ -0,0 +1,16 @@
+package com.contentgrid.appserver.rest.exception;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+public class InvalidRelationTargetException extends Exception {
+    @Getter
+    @NonNull
+    private final String reference;
+
+    public InvalidRelationTargetException(@NonNull String reference) {
+        super("Invalid relation target: '%s'".formatted(reference));
```

Comment:
I usually prefer to override `getMessage()` instead of pre-formatting the message and passing it to the exception.

The reason for that is so exceptions that *extend* this one can easily provide their own message